storio-sqlite-annotations-processor/src/main/java/com/pushtorefresh/storio/sqlite/annotations/processor/generate/TableGenerator.java

```diff
@@ -0,0 +1,138 @@
+
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+import javax.lang.model.element.Modifier;
+
+import static com.pushtorefresh.storio.common.annotations.processor.generate.Common.INDENT;
+
+/**
+ * Created by Pawel Bochenski on 11.12.2016.
+ */
+
+public class TableGenerator implements Generator<StorIOSQLiteTypeMeta> {
+
+    private static final String SUFFIX = "Table";
+    private static final String CREATE_METHOD_NAME = "createTable";
+    private static final String UPDATE_METHOD_NAME = "updateTable";
+    private static final String UPDATE_METHOD_VERSION_OLD_PARAM = "versionOld";
+    private static final String METHOD_DB_PARAM = "db";
+
+    private final ClassName db = ClassName.get("android.database.sqlite", "SQLiteDatabase");
+    private static final String tableName = "tableName";
+
+    @NotNull
+    @Override
+    public JavaFile generateJavaFile(@NotNull StorIOSQLiteTypeMeta typeMeta) {
+        TypeSpec tableSpec = TypeSpec.classBuilder(typeMeta.simpleName + SUFFIX)
+                .addFields(generateFields(typeMeta.storIOType.table(), typeMeta.columns.values()))
+                .addMethod(generateCreateMethod(typeMeta.storIOType.table(), typeMeta.columns.values()))
+                .addMethod(generateUpdateMethod(typeMeta.storIOType.table(), typeMeta.columns.values()))
+                .addModifiers(Modifier.PUBLIC, Modifier.FINAL)
+                .addMethod(generateConstructor())
+                .build();
+        return JavaFile.builder(typeMeta.packageName, tableSpec)
+                .indent(INDENT)
+                .build();
+    }
+
+    private Iterable<FieldSpec> generateFields(String table, Collection<StorIOSQLiteColumnMeta> columns) {
+        List<FieldSpec> list = new ArrayList<FieldSpec>();
+        list.add(FieldSpec.builder(String.class, tableName)
+                .initializer("$S", table)
+                .addModifiers(Modifier.PUBLIC, Modifier.STATIC, Modifier.FINAL)
+                .build());
+        for (StorIOSQLiteColumnMeta m : columns) {
+            list.add(FieldSpec.builder(String.class, "COLUMN_" + m.elementName)
+                    .initializer("$S", m.storIOColumn.name())
+                    .addModifiers(Modifier.PUBLIC, Modifier.STATIC, Modifier.FINAL)
+                    .build());
+        }
+        return list;
+    }
+
+    private MethodSpec generateConstructor() {
+        return MethodSpec.constructorBuilder().addModifiers(Modifier.PRIVATE).build();
+    }
+
+    private MethodSpec generateCreateMethod(String table, Collection<StorIOSQLiteColumnMeta> columns) {
+        String code = "CREATE TABLE " + table + " (";
+        int i = 0;
+        for (StorIOSQLiteColumnMeta entry : columns) {
+            code += entry.storIOColumn.name() + " " + entry.javaType.getSqlType();
+            if (!entry.storIOColumn.ignoreNull()) {
```

Comment:
=( `ignoreNull` absence doesn't guarantee database value existence.
It's just prevent overriding db field of there is no value in java entity.
Can we check `NotNull`, `NonNull` annotations?